include/dali/core/small_vector.h

```diff
@@ -248,6 +248,10 @@ class SmallVector : SmallVectorAlloc<T, allocator>, SmallVectorBase<T> {
   DALI_NO_EXEC_CHECK
   DALI_HOST_DEV void copy_assign(const T *data, size_t count) {
     clear();
+    if (!data) {
+      assert(count == 0);
+      return;
+    }
```

Comment:
This is not necessary and `assert` doesn't come for free in CUDA code (it significantly increases register pressure) and SmallVector can (at least theoretically) be used in device-side code.
Also, the logic is somewhat broken in non-debug builds - there's no assert and the function will quit silently instead of reproducibly causing an access violation and terminating. This might expose the calling code to security threats by hiding the situation where the vector contents were expected to be replaced but instead remained unchanged due to null source pointer.
Please revert this change.

Reply:
A test passes null  begin and end pointers to SmallVector. So we should check it to avoid calling memcpy with null pointers.